bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,907 @@
+                updateItemsFromPanelSettings();
+                if (currentState.isPowerlinkMode()) {
+                    logger.info("PowerMax alarm binding: running in Powerlink mode");
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.RESTORE);
+                } else {
+                    logger.info("PowerMax alarm binding: running in Standard mode");
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.ZONESNAME);
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.ZONESTYPE);
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.STATUS);
+                }
+                PowerMaxCommDriver.getTheCommDriver().exitDownload();
+            }
+        }
+    }
+
+    /**
+     * Post item updates on the bus to match a new alarm system state
+     *
+     * @param selector
+     *            filter on selector type; if null, no filter on selector type
+     * @param state
+     *            the alarm system state
+     */
+    private void updateItemsFromAlarmState(PowerMaxSelectorType selector, PowerMaxState state) {
+        updateItemsFromAlarmState(null, null, selector, state);
+    }
+
+    /**
+     * Post item updates on the bus to match a new alarm system state
+     *
+     * @param state
+     *            the alarm system state
+     */
+    private void updateItemsFromAlarmState(PowerMaxState state) {
+        updateItemsFromAlarmState(null, null, null, state);
+    }
+
+    /**
+     * Post item updates on the bus to match a new alarm system state
+     *
+     * @param provider
+     *            filter on provider; if null, no filter on provider
+     * @param name
+     *            filter on item name; if null, no filter on item name
+     * @param selector
+     *            filter on selector type; if null, no filter on selector type
+     * @param state
+     *            the alarm system state
+     */
+    private synchronized void updateItemsFromAlarmState(PowerMaxBindingProvider provider, String name,
+            PowerMaxSelectorType selector, PowerMaxState state) {
+
+        if (state == null) {
+            return;
+        }
+        if (itemUpdateDisabled) {
+            logger.debug("updateItemsFromAlarmState(): items update disabled");
+            return;
+        }
+
+        if (provider == null) {
+            for (PowerMaxBindingProvider prov : providers) {
+                updateItemsFromAlarmState(prov, name, selector, state);
+            }
+        } else {
+            for (String itemName : provider.getItemNames()) {
+                if ((name == null) || itemName.equals(name)) {
+                    State itemState = null;
+
+                    PowerMaxBindingConfig config = provider.getConfig(itemName);
+                    Integer num = config.getSelectorIntParam();
+                    Item item = config.getItem();
```

Comment:
This is where you would be concerned with acceptedDataTypes saved in the config instead of the item, and rework the following code to simply satisfy a type on the list instead of being concerned with item types, keeping of the knowledge of item->type mapping out of the binding.